src/relations/minio.py

```diff
@@ -0,0 +1,137 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Define the Airbyte server minio relation."""
+
+import logging
+
+from charmed_kubeflow_chisme.exceptions import ErrorWithStatus
+from ops import framework
+from ops.model import BlockedStatus, WaitingStatus
+from serialized_data_interface import (
+    NoCompatibleVersions,
+    NoVersionsListed,
+    get_interfaces,
+)
+
+from charm_helpers import construct_svc_endpoint
+from log import log_event_handler
+
+logger = logging.getLogger(__name__)
+
+
+class MinioRelation(framework.Object):
+    """Client for airbyte:minio relation."""
+
+    def __init__(self, charm):
+        """Construct.
+
+        Args:
+            charm: The charm to attach the hooks to.
+        """
+        super().__init__(charm, "object-storage")
+        self.charm = charm
+
+        # Handle minio relation.
+        charm.framework.observe(charm.on.object_storage_relation_joined, self._on_object_storage_relation_changed)
+        charm.framework.observe(charm.on.object_storage_relation_changed, self._on_object_storage_relation_changed)
+        charm.framework.observe(charm.on.object_storage_relation_broken, self._on_object_storage_relation_broken)
+
+    @log_event_handler(logger)
+    def _on_object_storage_relation_changed(self, event):
+        """Handle changing object-storage relation.
+
+        Args:
+            event: The event triggered when the relation changed.
+        """
+        if not self.charm.unit.is_leader():
+            return
+
+        if not self.charm._state.is_ready():
+            event.defer()
+            return
+
+        try:
+            interfaces = self._get_interfaces()
+            storage_data = self._get_object_storage_data(interfaces)
+            endpoint = construct_svc_endpoint(
+                storage_data["service"],
+                storage_data["namespace"],
+                storage_data["port"],
+                storage_data["secure"],
+            )
+
+            self.charm._state.minio = {
+                **storage_data,
+                "endpoint": endpoint,
+            }
+            self.charm._update(event)
```

Comment:
Python doesn't enforce the distinction between public and private member/method names.

Then again this very PR creates this member and method.

Would it make sense to make them "public", i.e. `self.charm.state` and `self.charm.update()`?